File: for-local-test/168.excel-sheet-column-title.java
class Solution {
    public String convertToTitle(int columnNumber) {
        if(columnNumber == 0){
            return "";
        }
        StringBuilder sb = new StringBuilder();
        while(columnNumber > 0){
            sb.append((char)('A' + (columnNumber - 1) % 26));
            columnNumber = (columnNumber - 1) / 26;
        }
        return sb.reverse().toString();
    }
    public static void main(String[] args) {
        Solution solution = new Solution();
        int columnNumber = 22;
        String title = solution.convertToTitle(columnNumber);
        System.out.println("Column number: " + columnNumber + ", Column title: " + title);
    }
}

File: for-local-test/python/168.excel-sheet-column-title.py
class Solution:
    def convertToTitle(self, columnNumber: int) -> str:
        if columnNumber == 0:
            return ""
        result = []
        while columnNumber > 0:
            result.append(chr(ord('A') + (columnNumber - 1) % 26))
            # Reason for use // instead of / is that we want to get the quotient as an integer, not a float
            # in python 3, the / operator always returns a float, even if both operands are integers. 
            # The // operator performs floor division, which means it returns the largest integer less than or equal to the result of the division.
            columnNumber = (columnNumber - 1) // 26
        return "".join(reversed(result))
    
if __name__ == "__main__":
    solution = Solution()
    columnNumber = 28
    title = solution.convertToTitle(columnNumber)
    print(f"Column number: {columnNumber}, Column title: {title}")  # Output: Column number: 28, Column title: AB